### PR TITLE
Allow searching by Issue Index

### PR DESCRIPTION
**Pull request by 6543**
commits taken from #17929

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
mysql integration test fails - could be an elastic search related issue :/

**Comment by 6543** [comment]
problem: test dont show errors!!! (https://drone.gitea.io/go-gitea/gitea/54666/2/14)

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
Isn't we can get of issues via index from databases? Why we need this PR?

**Comment by wxiaoguang** [comment]
> Isn't we can get of issues via index from databases? Why we need this PR?

At that time (I reviewed the original PR), I think that changing the search logic is more heavy than adding a search field.

For example, if you input "12345", it might be an issue index, it might be an keyword to be search, putting it into the search engine is quite simple.

Indeed, this PR is a nice to have, not a must. We can teach users to input the issue index in browser's address bar directly too, it depends how we think. So, merging it or not, either is fine for me.

File: modules/indexer/issues/bleve.go [hunk 1]
```diff
@@ -244,6 +247,7 @@ func (b *BleveIndexer) Search(ctx context.Context, keyword string, repoIDs []int
 	for _, repoID := range repoIDs {
 		repoQueriesP = append(repoQueriesP, numericEqualityQuery(repoID, "RepoID"))
 	}
+	index, _ := strconv.ParseInt(keyword, 10, 64)
```
Comment (lunny) [inline modules/indexer/issues/bleve.go:250]: Maybe we need to check if index > 0 to decide to if we need search the field.

**Comment by lunny** [comment]
When you input `#123` or `a/b#123` in the search box, we could directly go to issue 123. But when you input `123`, should we search the issue with index 123? I think it's annoying and unnecessary.

**Comment by wxiaoguang** [comment]
> When you input `#123` or `a/b#123` in the search box, we could directly go to issue 123. But when you input `123`, should we search the issue with index 123? I think it's annoying and unnecessary.

Yep, that's also a solution. The PR author didn't accept it before. Refer to the original PR, it has been discussed.

**Comment by wxiaoguang** [comment]
Since there is no new requirement for this feature, and there is still divergence on it.

I would propose to convert it to a draft, leave it as it is, until there is some new thoughts or new requirements about it. (maybe by some RFC in the future).

I don't think we should merge it just because someone has approved it at the moment.

**Comment by wxiaoguang** [comment]
Will be replaced by #24479

**Pull request closed without merging**